src/main/g8/src/main/scala/com/example/WebServer.scala

```diff
@@ -0,0 +1,38 @@
+package com.example
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ ContentTypes, HttpEntity }
+import akka.http.scaladsl.server.Directives
+import akka.stream.ActorMaterializer
+
+import scala.io.StdIn
+
+object WebServer extends Directives {
+  def main(args: Array[String]) {
+
+    implicit val system = ActorSystem("my-system")
+    implicit val materializer = ActorMaterializer()
+    // needed for the future flatMap/onComplete in the end
+    implicit val executionContext = system.dispatcher
+
+    val bindingFuture = Http().bindAndHandle(route, "localhost", 8080)
+
+    println(s"Server online at http://localhost:8080/\nPress RETURN to stop...")
+    StdIn.readLine() // let it run until user presses return
+    bindingFuture
+      .flatMap(_.unbind()) // trigger unbinding from the port
+      .onComplete(_ => system.terminate()) // and shutdown when done
+  }
+
+  // Here you can define all the different routes you want to have served by this web server.
+  val route = pathEndOrSingleSlash { // Listens to the top `/`
```

Comment:
What do you think about putting the route into an extra file right from the beginning (at least for this style). This would show that routes are values that can come from everywhere and would separate the boot code from the handling code.